### PR TITLE
daggy: migrate to Conan v2

### PR DESCRIPTION
**Pull request by valgur**

### CODE REVIEW

File: recipes/daggy/2.1/conanfile.py [hunk 1]
```diff
@@ -1,95 +1,73 @@
...
-                    self.settings.compiler, 
-                    self.settings.compiler.version))
+    def export_sources(self):
+        export_conandata_patches(self)
 
-        if not self.options["qt"].shared: 
-            raise ConanInvalidConfiguration("Shared Qt lib is required.") 
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
```
Comment (CJCombrink) [inline recipes/daggy/2.1/conanfile.py:58]: This should probably be
```
self.options.rm_safe("fPIC")
```

**Review by jwillikers — APPROVED** [review]

File: recipes/daggy/2.1/conanfile.py [hunk 2]
```diff
@@ -1,138 +1,135 @@
...
+    def generate(self):
+        env = VirtualBuildEnv(self)
+        env.generate()
+
+        # Required for Qt's moc
+        env = VirtualRunEnv(self)
+        env.generate(scope="build")
+
+        tc = CMakeToolchain(self)
+        tc.variables["CMAKE_CXX_STANDARD"] = 17
```
Comment (RubenRBS) [inline recipes/daggy/2.1/conanfile.py:107]: This seems a bit weird. Why is it forcing C++ 17? If this were necessary for the package, we should either add max_checks in the validate or erase the standard in the package_info methods

**Review by RubenRBS — COMMENTED** [review]
Thanks! Sorry for the delay on this one

I have a question regarding the `CMAKE_CXX_STANDARD` cmake variable, otherwise looks good

**Comment by RubenRBS** [comment]
When locally building with `$ conan create . --version=2.1.3 -b=missing -o="&:with_ssh2=False" -o="&:with_yaml=False"`

I get
```

======== Exporting recipe to the cache ========
daggy/2.1.3: Exporting package recipe: /Users/ruben/coding/support/external/valgur/recipes/daggy/2.1/conanfile.py
daggy/2.1.3: exports: File 'conandata.yml' found. Exporting it...
daggy/2.1.3: Copied 1 '.py' file: conanfile.py
daggy/2.1.3: Copied 1 '.yml' file: conandata.yml
daggy/2.1.3: Exported to cache folder: /Users/ruben/.conan2/p/daggy2de924c2f4119/e
daggy/2.1.3: Exported: daggy/2.1.3#7384df7a7b2a8314918146a03d985a3a (2024-04-03 11:30:06 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos
[options]
&:with_ssh2=False
&:with_yaml=False
[platform_tool_requires]
cmake/3.26.4

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=gnu17
compiler.libcxx=libc++
compiler.version=15
os=Macos
[platform_tool_requires]
cmake/3.26.4


...


-------- Installing package daggy/2.1.3 (10 of 10) --------
daggy/2.1.3: Building from source
daggy/2.1.3: Package daggy/2.1.3:441b48a735cf79424618a35ba28390bf7aa35610
daggy/2.1.3: Copying sources to build folder
daggy/2.1.3: Building your package in /Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b
daggy/2.1.3: Calling generate()
daggy/2.1.3: Generators folder: /Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/build/Release/generators
daggy/2.1.3: CMakeToolchain generated: conan_toolchain.cmake
daggy/2.1.3: Requested parallel jobs: 12
daggy/2.1.3: CMakeToolchain generated: CMakePresets.json
daggy/2.1.3: CMakeToolchain generated: ../../../src/CMakeUserPresets.json
daggy/2.1.3: CMakeDeps necessary find_package() and targets for your CMakeLists.txt
    find_package(Qt6)
    find_package(kainjow_mustache)
    target_link_libraries(... qt::qt kainjow_mustache::kainjow_mustache)
daggy/2.1.3: Generating aggregated env files
daggy/2.1.3: Generated aggregated env files: ['conanbuild.sh']
daggy/2.1.3: Calling build()
daggy/2.1.3: Running CMake.configure()
daggy/2.1.3: RUN: cmake -G "Unix Makefiles" -DCMAKE_TOOLCHAIN_FILE="/Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/build/Release/generators/conan_toolchain.cmake" -DCMAKE_INSTALL_PREFIX="/Users/ruben/.conan2/p/b/daggyb397d7dd2671b/p" -DCMAKE_POLICY_DEFAULT_CMP0091="NEW" -DCMAKE_BUILD_TYPE="Release" "/Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/src/src"
-- Using Conan toolchain: /Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/build/Release/generators/conan_toolchain.cmake
-- Conan toolchain: Setting CMAKE_POSITION_INDEPENDENT_CODE=ON (options.fPIC)
-- Conan toolchain: C++ Standard 17 with extensions ON
-- Conan toolchain: Setting BUILD_SHARED_LIBS = OFF
-- The CXX compiler identification is AppleClang 15.0.0.15000309
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Git: /usr/bin/git (found version "2.39.3 (Apple Git-146)")
fatal: not a git repository (or any of the parent directories): .git
-- Conan: Component target declared 'Qt6::Core'
-- Conan: Component target declared 'Qt6::Platform'
-- Conan: Component target declared 'Qt6::Gui'
-- Conan: Component target declared 'Qt6::QCocoaIntegrationPlugin'
-- Conan: Component target declared 'qt::QCocoaIntegrationPlugin'
-- Conan: Component target declared 'Qt6::QGifPlugin'
-- Conan: Component target declared 'Qt6::QIcoPlugin'
-- Conan: Component target declared 'Qt6::QSQLiteDriverPlugin'
-- Conan: Component target declared 'Qt6::QPSQLDriverPlugin'
-- Conan: Component target declared 'Qt6::QODBCDriverPlugin'
-- Conan: Component target declared 'qt::QODBCDriverPlugin'
-- Conan: Component target declared 'Qt6::Network'
-- Conan: Component target declared 'Qt6::Sql'
-- Conan: Component target declared 'Qt6::Test'
-- Conan: Component target declared 'Qt6::Widgets'
-- Conan: Component target declared 'Qt6::PrintSupport'
-- Conan: Component target declared 'Qt6::OpenGL'
-- Conan: Component target declared 'Qt6::OpenGLWidgets'
-- Conan: Component target declared 'Qt6::Concurrent'
-- Conan: Component target declared 'Qt6::Xml'
-- Conan: Target declared 'qt::qt'
-- Conan: Component target declared 'odbc::_odbc'
-- Conan: Component target declared 'odbc::odbcinst'
-- Conan: Component target declared 'odbc::odbccr'
-- Conan: Target declared 'ODBC::ODBC'
-- Conan: Target declared 'libtool::libtool'
-- Conan: Target declared 'automake::automake'
-- Conan: Target declared 'autoconf::autoconf'
-- Conan: Target declared 'm4::m4'
-- Conan: Target declared 'opengl::opengl'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Core/conan_qt_executables_variables.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Core/conan_qt_qt6_coreprivate.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Core/Qt6CoreMacros.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Core/Qt6CoreConfigExtras.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/Qt6ConfigExtras.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicExternalProjectHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicCMakeVersionHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicCMakeHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicDependencyHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicAppleHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicTargetHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicFinalizerHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicTestHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicWalkLibsHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicToolHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicPluginHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicFindPackageHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6/QtPublicWasmToolchainHelpers.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Gui/conan_qt_qt6_guiprivate.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Widgets/conan_qt_qt6_widgetsprivate.cmake'
-- Conan: Including build module from '/Users/ruben/.conan2/p/b/qtc0235a8a92fd4/p/lib/cmake/Qt6Widgets/Qt6WidgetsMacros.cmake'
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring done (0.9s)
-- Generating done (0.0s)
-- Build files have been written to: /Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/build/Release

daggy/2.1.3: Running CMake.build()
daggy/2.1.3: Requested parallel jobs: 12
daggy/2.1.3: RUN: cmake --build "/Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/build/Release" -- -j12
[  0%] Built target DaggyCore_autogen_timestamp_deps
[  6%] Automatic MOC for target DaggyCore
[  6%] Built target DaggyCore_autogen
[ 12%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/cmake_pch_arm64.hxx.pch
[ 18%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/DaggyCore_autogen/mocs_compilation.cpp.o
[ 25%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/Core.cpp.o
[ 31%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/Sources.cpp.o
[ 37%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/Errors.cpp.o
[ 43%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/providers/IFabric.cpp.o
[ 50%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/aggregators/IAggregator.cpp.o
[ 56%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/aggregators/CConsole.cpp.o
[ 62%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/providers/CLocal.cpp.o
[ 68%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/providers/IProvider.cpp.o
[ 75%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/aggregators/CFile.cpp.o
[ 81%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/providers/CLocalFabric.cpp.o
[ 87%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/core_ansi_c.cpp.o
/Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/src/src/DaggyCore/Sources.cpp:374:59: error: 'yaml' is missing exception specification 'noexcept'
std::optional<daggy::Sources> daggy::sources::convertors::yaml(const QString& data, QString& error)
                                                          ^
                                                                                                    noexcept
/Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/src/src/DaggyCore/Sources.hpp:91:41: note: previous declaration is here
DAGGYCORE_EXPORT std::optional<Sources> yaml(const QString& data, QString& error) noexcept;
                                        ^
[ 93%] Building CXX object DaggyCore/CMakeFiles/DaggyCore.dir/aggregators/CCallback.cpp.o
1 error generated.
make[2]: *** [DaggyCore/CMakeFiles/DaggyCore.dir/Sources.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [DaggyCore/CMakeFiles/DaggyCore.dir/all] Error 2
make: *** [all] Error 2

daggy/2.1.3: ERROR: 
Package '441b48a735cf79424618a35ba28390bf7aa35610' build failed
daggy/2.1.3: WARN: Build folder /Users/ruben/.conan2/p/b/daggyb397d7dd2671b/b/build/Release
ERROR: daggy/2.1.3: Error in build() method, line 125
	cmake.build()
	ConanException: Error 2 while executing
```

**Comment by valgur** [comment]
@RubenRBS Thanks!
Unrelated, but what does the `&` ampersand in `-o="&:with_ssh2=False"` mean? I don't remember seeing it in the docs and it's kind of impossible to google.

**Comment by RubenRBS** [comment]
> Unrelated, but what does the & ampersand in -o="&:with_ssh2=False" mean? I don't remember seeing it in the docs and it's kind of impossible to google.

We'll make sure to make it a bit more prominent in the docs ([It's documented in here, but just barely](https://docs.conan.io/2/reference/config_files/profiles.html#profile-patterns))

It means "only affect the package I'm creating" - in this case it would be equivalent to `daggy/1.2.3:with_ssh2=False`, but it's faster to type and works when the package has no name :)

**Comment by RubenRBS** [comment]
@valgur ping on the failing compilation configuration

**Comment by RubenRBS** [comment]
Looks good, will doublecheck upstream about the yaml issue before proceeding though :)

**Pull request closed without merging**